File: bindings/perl/examples/example2.pl
#!/usr/bin/perl

# SPDX-FileCopyrightText: 2006-2021 Istituto Italiano di Tecnologia (IIT)
# SPDX-FileCopyrightText: 2006-2010 RobotCub Consortium
# SPDX-License-Identifier: BSD-3-Clause

use strict;
use warnings;

use yarp;

# set up a description of the device we want - just the fake framegrabber
my $prop = new yarp::Property();
$prop->put("device","fakeFrameGrabber");
$prop->put("width", 640);
$prop->put("height", 480);
print "Property is ", $prop->toString(), "\n";

# create the device
my $dev = new yarp::PolyDriver($prop);

# get the "IFrameGrabberImage" interface
my $grabber = $dev->viewIFrameGrabberImage();

# grab 30 images
my $img = new yarp::ImageRgb();
for (my $i=0; $i<30; $i++) {
    $grabber->getImage($img);
    my $w = $img->width();
    my $h = $img->height();
    print "Got a ${w}x${h} image\n";
}

# shut things down
$dev->close();

File: bindings/lua/examples/example2.lua
#!/usr/bin/lua

-- SPDX-FileCopyrightText: 2006-2021 Istituto Italiano di Tecnologia (IIT)
-- SPDX-License-Identifier: BSD-3-Clause


-- LUA_CPATH should have the path to yarp-lua binding library (i.e. yarp.so, yarp.dll)
require("yarp")

-- initialize yarp network
yarp.Network()


-- call static member in this way
-- t = yarp.Time_now()

-- create and open sender/receiver ports
sender = yarp.BufferedPortBottle()
receiver = yarp.BufferedPortBottle()
sender:open("/lua/write")
receiver:open("/lua/read")

-- connect sender to receiver
yarp.NetworkBase_connect(sender:getName(), receiver:getName())

for i=1,10 do

    -- write to the sender port
    local wb = sender:prepare()
    wb:clear()
    wb:addString("count")
    wb:addInt32(i)
    wb:addString("of")
    wb:addInt32(10)
    sender:write()

    -- read from receiver port
    local rb = receiver:read();
    print("Received: ", rb:toString())

    -- wait for a second
    yarp.delay(1.0)
end

-- disconnect sender from receiver
yarp.NetworkBase_disconnect(sender:getName(), receiver:getName())

-- close the ports
sender:close()
receiver:close()

-- Deinitialize yarp network
yarp.Network_fini()
